### PR TITLE
Add solution to 406 | Queue Reconstruction by Height

### PR DESCRIPTION
**Pull request by jarvisdev**

### CODE REVIEW

File: C++/queue_reconstruction_by_height.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+bool comp(vector<int> a, vector<int> b) {
+    return a[0]<b[0];
+}
+class Solution {
+public:
+    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
+        vector<vector<int>> ret(people.size());
+        sort(people.begin(),people.end(),comp);
+        for(vector<int> v: people) {
+            cout<<v[0]<<" "<<v[1]<<endl;
```
Comment (noisefilter19) [inline C++/queue_reconstruction_by_height.cpp:10]: Do we need to print this?

File: C++/queue_reconstruction_by_height.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            int x = v[1];
+            for(int i=0;i<ret.size();i++) {
+                if(ret[i].size()==0 || ret[i][0]==v[0]) {
```
Comment (noisefilter19) [inline C++/queue_reconstruction_by_height.cpp:13]: Is the 2nd condition enough or should we include ret[i][0]>=v[0]? Please add a comment explaining this as well.

**Review by noisefilter19 — CHANGES_REQUESTED** [review]

File: C++/queue_reconstruction_by_height.cpp (hunk 1 above)
Comment (jarvisdev) [reply C++/queue_reconstruction_by_height.cpp:10]: Nope removing this.

File: C++/queue_reconstruction_by_height.cpp (hunk 2 above)
Comment (jarvisdev) [reply C++/queue_reconstruction_by_height.cpp:13]: making second condition `>=` would be redundant as we have sorted the array before so in ret array only those pairs will be there which have height <= the height of current pair so we just need to check for == here

Comment (noisefilter19) [reply C++/queue_reconstruction_by_height.cpp:13]: Cool!

**Review by noisefilter19 — APPROVED** [review]

**Pull request merged**